src/Colorfy.jl

```diff
@@ -87,13 +87,35 @@ defaultscheme(values) = colorschemes[:viridis]
 
 Colors mapped from the `colorfier` .
 """
-colors(colorfier::Colorfier) = coloralpha.(getcolors(colorfier), alphas(colorfier))
+function colors(colorfier::Colorfier)
+  # find invalid indices
+  isinvalid(v) = ismissing(v) || (v isa Number && !isfinite(v))
+  vals = values(colorfier)
+  iinds = findall(isinvalid, vals)
+
+  if isempty(iinds)
+    coloralpha.(getcolors(colorfier), alphas(colorfier))
+  else
+    # invalid values are assigned full transparency
+    vcolors = Vector{Colorant}(undef, length(vals))
+    vcolors[iinds] .= colorant"transparent"
+    
+    # set colors of valid values
+    vinds = setdiff(1:length(vals), iinds)
```

Comment:
move this up after the definition of iinds

Reply:
This line is only used in the else branch, it is not necessary to move to the top level.